lib/util/calculation.dart

```diff
@@ -0,0 +1,281 @@
+library calculation;
+
+import 'dart:math';
+
+import 'dartulator_strings.dart';
+
+enum Precedence { i, ii, iii, iv, v }
+
+abstract class Calculation implements Function {
+  static bool usingDegrees = false;
+
+  final Precedence _precedenceLevel;
+
+  const Calculation(this._precedenceLevel);
+
+  bool hasLessPrecedenceThan(Calculation other) {
+    return _precedenceLevel.index < other._precedenceLevel.index;
+  }
+
+  @override
+  String toString() => "";
+
+// for the arc(sine cosine tangent) functions to switch the answer to degrees if need be
+  static Function arcTrigWrapper(num arcTrigFunction(x)) {
+    num _convertedArcTrigFunction(num x) {
+      num value = arcTrigFunction(x);
+      if (usingDegrees) {
+        value *= 180 / PI;
+      }
+      return value;
+    }
+    return _convertedArcTrigFunction;
+  }
+
+// for the sine cosine tangent functions to switch the answer to degrees if need be
+  static Function trigWrapper(num trigFunction(x)) {
+    num _convertedTrigFunction(num x) {
+      if (usingDegrees) {
+        x *= PI / 180;
+      }
+      return trigFunction(x);
+    }
+    return _convertedTrigFunction;
+  }
+}
+
+abstract class UnaryCalculation extends Calculation {
+  const UnaryCalculation(Precedence levelOfImportance)
+      : super(levelOfImportance);
+
+  num call(num x);
+}
+
+abstract class BinaryCalculation extends Calculation {
+  const BinaryCalculation(Precedence levelOfImportance)
+      : super(levelOfImportance);
+
+  num call(num x, num y);
+}
+
+class BasicCalculation {}
```

Comment:
When doing different checks in the app_component.dart file We needed to know which calculations were Basic or not to find out if a move was valid. More like this one are probably coming soon.